pkg/resource/kubeconfig_factory.go

```diff
@@ -20,19 +23,42 @@ func (r *Helper) getKubeconfigFactory(namespace string) (cmdutil.Factory, error)
 		overrides.Context.Namespace = namespace
 	}
 	clientConfig := clientcmd.NewNonInteractiveClientConfig(*config, "", overrides, nil)
+	restConfig, err := clientConfig.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	if r.metricsEnabled {
+		restConfig.WrapTransport = func(rt http.RoundTripper) http.RoundTripper {
```

Comment:
See comment for restconfig_factory.go
This is not an issue as long as we're using certificate-based auth as returned by the installer. But if at some point we switch to a service account + token, then you will likely have a WrapTransport already set.